x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DayName.java

```diff
@@ -0,0 +1,145 @@
+    )
+    public DayName(
+        Source source,
+        @Param(
+            name = "date",
+            type = { "date", "date_nanos" },
+            description = "Date expression. If `null`, the function returns `null`."
+        ) Expression date,
+        Configuration configuration
+    ) {
+        super(source, List.of(date), configuration);
+        this.field = date;
+    }
+
+    private DayName(StreamInput in) throws IOException {
+        this(Source.readFrom((PlanStreamInput) in), in.readNamedWriteable(Expression.class), ((PlanStreamInput) in).configuration());
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        source().writeTo(out);
+        out.writeNamedWriteable(field);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
+    }
+
+    Expression field() {
+        return field;
+    }
+
+    @Override
+    public DataType dataType() {
+        return DataType.KEYWORD;
+    }
+
+    @Override
+    protected TypeResolution resolveType() {
+        if (childrenResolved() == false) {
+            return new TypeResolution("Unresolved children");
+        }
+
+        String operationName = sourceText();
+        TypeResolution resolution = TypeResolutions.isType(field, DataType::isDate, operationName, FIRST, "datetime or date_nanos");
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+
+        return TypeResolution.TYPE_RESOLVED;
+    }
+
+    @Override
+    public boolean foldable() {
+        return field.foldable();
+    }
+
+    @Override
+    public EvalOperator.ExpressionEvaluator.Factory toEvaluator(ToEvaluator toEvaluator) {
+        var fieldEvaluator = toEvaluator.apply(field);
+        return getEvaluator(field().dataType(), fieldEvaluator);
+    }
+
+    private EvalOperator.ExpressionEvaluator.Factory getEvaluator(
+        DataType dateType,
+        EvalOperator.ExpressionEvaluator.Factory fieldEvaluator
+    ) {
+        if (dateType == DataType.DATE_NANOS) {
+            return new DayNameNanosEvaluator.Factory(source(), fieldEvaluator, configuration().zoneId(), configuration().locale());
+        }
+        return new DayNameMillisEvaluator.Factory(source(), fieldEvaluator, configuration().zoneId(), configuration().locale());
+    }
+
+    @Override
+    public Expression replaceChildren(List<Expression> newChildren) {
+        return new DayName(source(), newChildren.get(0), configuration());
+    }
+
+    @Override
+    protected NodeInfo<? extends Expression> info() {
+        return NodeInfo.create(this, DayName::new, field, configuration());
+    }
+
+    @Evaluator(extraName = "Millis")
+    static BytesRef processMillis(long val, @Fixed ZoneId zoneId, @Fixed Locale locale) {
+        ZonedDateTime dateTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(val), zoneId);
+        String displayName = dateTime.getDayOfWeek().getDisplayName(TextStyle.FULL, locale);
```

Comment:
@nik9000 I implemented taking into consideration the locale and zone of the instance.

Is this how elastic usually behaves on locale specific content or do we default to English always? How is this configured?